spec/rspec/core/example_group_spec.rb

```diff
@@ -140,6 +140,12 @@ def metadata_hash(*args)
         expect(child).to have_class_const("SomeParentGroup::Hash")
       end
 
+      it 'removes it\'s constants' do
+        groups = RSpec::ExampleGroups
+        groups.remove_all_constants
```

Comment:
`remove_all_constants` is not part of RSpec's public API, and as such, it's really just an implementation detail.  The thing we want to test is the behavior of `RSpec.world.reset` and `RSpec.reset`, since those are the public APIs that we are improving in this PR.